pyquil/quil.py

```diff
@@ -728,8 +734,7 @@ def pop(self) -> AbstractInstruction:
         Removes the last instruction from the program and returns it
         """
         last = self.instructions[-1]
-        instructions = self._program.to_headers() + self.instructions
-        new_program = Program(instructions[:-1])
+        new_program = Program(self.instructions[:-1])
```

Comment:
I guess the catch here is that this will return `None` now if the program is only "headers"? This is where that `iter_instructions` method might come in handy, `program.iter_instructions().rev().next()` could do part of the job here, although removing it would be a different problem.

Is this method used and do we need it? I see no usages in `pyQuil` itself, not even in tests. An audit of private code just now suggests it can be removed. It's simple enough to restore if needed, but let's break this.

Reply:
Actually, no catch. `self.instructions` is referencing the `instructions @property` on the `pyQuil` `Program`, which calls `quil-rs`' `to_instructions()`, which will return "headers" as instructions.

That said, I removed `pop` because it is hardly used anywhere. Technically it was used in one test, but only to get the last instruction so it was a simple fix.